RecoverCRPVM/CreateCRPRescueVM.ps1

```diff
@@ -0,0 +1,260 @@
+# Get running path
+$RunPath = split-path -parent $MyInvocation.MyCommand.Source
+cd $RunPath
+$CommonFunctions = $runPath+"\Common-Functions.psm1"
+
+#Import-Module Common-Functions -ArgumentList $LogFile -ErrorAction Stop 
+if (Get-Module Common-Functions) {remove-module -name Common-Functions}   
+Import-Module -Name $CommonFunctions  -ArgumentList $LogFile -ErrorAction Stop 
+write-log "Info: Log is being written to ==> $LogFile" 
+
+
+
+if (-not (Get-AzureRmContext).Account)
+{
+    Login-AzureRmAccount
+}
+
+#Set the context to the correct subid
+Write-Log "Setting the context to SubID $SubID" -color Yellow
+$subContext= Set-AzureRmContext -Subscription $SubID
+write-log $subContext -logOnly
+if ($subContext -eq $null) 
+{
+    Write-Log "Unable to set the Context for the given subId ==> $SubID, Please make sure you first  run the command ==> Login-AzureRMAccount" -Color Red
+    return
+}
+
+
+#Step 1 Get the VM Object
+Write-Log "Running Get-AzureRmVM -ResourceGroupName `"$ResourceGroup`" -Name `"$VmName`"" -Color Yellow
+$vm = Get-AzureRmVM -ResourceGroupName $ResourceGroup -Name $VmName 
+write-log "`"$vm`" => $($vm)" -logOnly
+
+if (-not (SupportedVM -vm $vm)) { return }
+Write-Log "Successfully got the VM Object info for $($vm.Name)" -Color Green
+if ($vm.StorageProfile.OsDisk.OsType -eq "Windows") 
+{
+    write-log "Detected 'Windows' as the OSType for $Vmname"      
+    $windowsVM= $true
+}
+else 
+{   
+    write-log "Detected 'Linux' as the OSType for $Vmname" 
+    $windowsVM= $false
+}
+
+#Step 2 Stop VM
+$stopped = StopTargetVM -ResourceGroup $ResourceGroup -VmName $VmName
+write-log "`"$stopped`" ==> $($stopped)" -logOnly
+if (-not $stopped) 
+{
+    Write-Log "Unable to stop the VM" -Color Red
+    Return
+}
+
+
+#Step 3 SnapshotAndCopyOSDisk  
+$osDiskVHDToBeRepaired = SnapshotAndCopyOSDisk -vm $vm -prefix $prefix
+write-log "`"$osDiskVHDToBeRepaired`" => $($osDiskVHDToBeRepaired)" -logOnly
+if (-not $osDiskVHDToBeRepaired)
+{
+    Write-Log "Unable to snapshot and copy the OS Disk to be repaired, cannot proceed" -Color Red
+    Return
+}
+
+#Step 4 Create Rescue VM
+$Vmname = $vm.Name
+$rescueVMNname = "$prefix$Vmname"
+$RescueResourceGroup = "$prefix$ResourceGroup"
+$rescueVm = CreateRescueVM -vm $vm -ResourceGroup $ResourceGroup  -rescueVMNname $rescueVMNname -RescueResourceGroup $RescueResourceGroup -prefix $prefix -Sku $sku -Offer $offer -Publisher $Publisher -Version $Version 
+Write-Log "$reccueVM ==> $($reccueVM)" -logOnly
+if (-not $rescuevm)
+{
+    Write-Log "Unable to create the Rescue VM, cannot proceed" -Color Red
+    Return
+}
+
+
+#Step 5 #Get a reference to the rescue VM Object
+Write-Log "Running Get-AzureRmVM -ResourceGroupName `"$RescueResourceGroup`" -Name `"rescueVMNname`"" -Color Yellow
+$rescuevm = Get-AzureRmVM -ResourceGroupName $RescueResourceGroup -Name $rescueVMNname
+if (-not $rescuevm)
```

Comment:
Would be better to add this validation in CreateRescueVM itself so that callers don't have to double-check